core/trino-main/src/main/java/io/trino/sql/planner/plan/TableScanNode.java

```diff
@@ -93,6 +98,26 @@ public TableScanNode(
         checkArgument(assignments.keySet().containsAll(outputs), "assignments does not cover all of outputs");
         this.enforcedConstraint = requireNonNull(enforcedConstraint, "enforcedConstraint is null");
         this.forDelete = forDelete;
+
+        if (!enforcedConstraint.isAll() && !enforcedConstraint.isNone()) {
+            Map<ColumnHandle, Domain> domains = enforcedConstraint.getDomains().orElseThrow();
+
+            Set<ColumnHandle> visibleColumns = outputs.stream()
+                    .map(assignments::get)
```

Comment:
let's also add a comment here that assignments can contain dangling columns (for now)

Reply:
That check verifies that assignments contains outputs, but assignments can still have more symbols in it than what output produces. So in that sense assignments can have symbol to column handle mappings that are no longer referenced. But as I mentioned in a comment above, this is needed for some obscure reasons in the EffectivePredicateExtractor.